catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaEditorEditText.java

```diff
@@ -289,10 +292,10 @@ public boolean undo() {
 		if (!history.undoIsPossible()) {
 			return false;
 		}
-		InternFormulaState lastStep = history.backward();
+		UndoState lastStep = history.backward();
 		if (lastStep != null) {
 
-			internFormula = lastStep.createInternFormulaFromState();
+			internFormula = lastStep.internFormulaState.createInternFormulaFromState();
```

Comment:
Is there a better way to call this function? What if lastStep.interFormulaState == null? Maybe it's possible to provide a function in the lastStep itself, so we don't have to dig into two layers.